docs/docs/the-component-lifecycle.md

```diff
@@ -0,0 +1,43 @@
+---
+id: the-component-lifecycle
+title: The Component Lifecycle
+permalink: docs/the-component-lifecycle.html
+prev: rendering-elements.html
+---
+
+Components have three main parts of their lifecycle:
+
+* **[Mounting](#mounting):** A component is being inserted into the DOM.
+* **[Updating](#updating):** A component is being re-rendered to determine if the DOM should be updated.
+* **[Unmounting](#unmounting):** A component is being removed from the DOM.
+
+React provides lifecycle methods that you can specify to hook into this process. Methods prefixed with **`will`** are called right before something happens, and methods prefixed with **`did`** are called right after something happens.
+
+## Mounting
+
+These methods are called when a component is being inserted into the DOM.
+
+* **`getInitialState(): object`** is invoked before a component is mounted. Stateful components should implement this and return the initial state data.
+* **`componentWillMount()`** is invoked immediately before mounting occurs.
+* **`componentDidMount()`** is invoked immediately after mounting occurs. Initialization that requires DOM nodes should go here.
+
+## Updating
+
+These methods are called when a component is being re-rendered and are used to determine if the DOM should be updated.
+
+* **`componentWillReceiveProps(object nextProps)`** is invoked when a mounted component receives new props. This method should be used to compare `this.props` and `nextProps` to perform state transitions using `this.setState()`.
+* **`shouldComponentUpdate(object nextProps, object nextState): boolean`** is invoked when a component decides whether any changes warrant an update to the DOM. Implement this as an optimization to compare `this.props` with `nextProps` and `this.state` with `nextState` and return `false` if React should skip updating.
```

Comment:
Here too, "Implement this as an optimization" sounds like you'd do this often but it can lead to nasty bugs when done incorrectly, and we almost never encourage people to implement it by hand. It's good to know it exists but we need to make it clear it's an escape hatch, and even if you think you need it, you're better off using `PureComponent` rather than writing it by hand.

Also, "warrant an update to the DOM" is technically incorrect. This is not what `shouldComponentUpdate` decides. `shouldComponentUpdate` prevents reconciling the whole tree below, so it's not just DOM: e.g. child components won't receive new props even if those values technically have changed.

Finally, we plan to relax `shouldComponentUpdate` to be a hint to React in the future. So it might not be respected. We need to make it clear so people don't rely on it to pause rendering e.g. while data is loading.

Maybe:

"`shouldComponentUpdate(...)` is lets the component provide a hint to React whether its or its children's output is affected by the current change in state or props. It defaults to `true`, and in the vast majority of cases you should rely on the default behavior. If you determine a specific component is slow after profiling, you may change it to inherit from `React.PureComponent` which implements `shouldComponentUpdate()` with a shallow prop and state comparison. If you are confident you want to write it by hand, you may compare `this.props` with `nextProps` and `this.state` with `nextState` and return `false` to tell React the update can be skipped. Returning `false` does not prevent child components from re-rendering when *their* state changes. Also note that React treats `shouldComponentUpdate()` as a hint rather than a strict directive, and may update your component anyway."